src/tools/gxpkg.ss

```diff
@@ -523,13 +592,91 @@
                  (error "Cannot build unknown package" pkg)))
             (build.ss (pkg-build-script pkg)))
        (displayln "... build " pkg)
+       (pkg-manifest! pkg)
        (run-process [build.ss "compile" build-options ...]
                     directory: path
                     coprocess: void
                     stdout-redirection: #f)
        (when dependents?
          (for-each pkg-build (pkg-dependents pkg)))))))
 
+(def (pkg-manifest! pkg)
+  (let* (((values pkg _) (pkg+tag pkg))
+         (plist (pkg-plist pkg))
+         (deps (pgetq depend: plist []))
+         (deps
+          (let recur ((rest deps) (result []))
+            (match rest
+              ([dep . rest]
+               (let ((values dep _) (pkg+tag dep))
+                 ;; check for external package manager installed deps (eg NiX)
+                 (if (file-exists? (pkg-plist-path dep))
+                   (let* ((plist (pkg-plist dep))
+                          (deps (pgetq depend: plist [])))
+                     (recur rest (recur deps (cons dep result))))
+                   ;; just record the dep, we don't have the pkg contents
+                   ;; for transitive
+                   (recur rest (cons dep result)))))
+              (else
+               (remove-duplicates result)))))
+         (manifests
+          (let lp ((rest deps) (result []))
+            (match rest
+              ([dep . rest]
+               (let (manifest
+                     (call-with-input-file
+                         (path-expand (string-append dep ".manifest")
```

Comment:
Wait, in the same directory as the source? What if it's read-only? That used to be possible, innit?

Reply:
input file

Reply:
it is one up in the pkg hierarchy, so we can write.